NineChronicles.Headless/GraphTypes/States/AgentStateType.cs

```diff
@@ -132,6 +133,25 @@ public AgentStateType()
                     context.Source.AgentAddress,
                     CrystalCalculator.CRYSTAL
                 ).GetQuantityString(true));
+            Field<NonNullGraphType<MeadPledgeType>>(
+                "pledge",
+                description: "mead pledge information.",
+                resolve: context =>
+                {
+                    var pledgeAddress = context.Source.AgentAddress.GetPledgeAddress();
+                    Address? address = null;
+                    bool approved = false;
+                    int mead = 0;
+                    if (context.Source.GetState(pledgeAddress) is List l)
+                    {
+                        address = l[0].ToAddress();
+                        approved = l[1].ToBoolean();
+                        mead = l[2].ToInteger();
+                    }
+
+                    return (address, approved, mead);
+                }
+            );
```

Comment:
As I know, there may be three states `NoRequest | WaitingApprove | ApprovedMeadPledge` (random naming 🙄) How about representing `NoRequest` as `null`? I wonder if there was another intention.

Reply:
like existing `activated` query, since it is used to determine information about the pledge contract state.